File: vue.jsonforms/src/entities/providers-and-introducers/IntroducerReferalProduct.ts
import { ScopeofReferalProduct } from "./ScopeofReferalProduct";

export class IntroducerReferalProduct {

    public isReferralsprovided: boolean = false;
    public fixedFeePerReferral: string = "";
    public fixedFeeAndAPercentageOfRevenue: string = "";
    public remunerationVaryByProduct: string = "";
    public remuneratedInitiallyOnRenewal: string = "";
    public monthsNotice: number = 0;
    public products: ScopeofReferalProduct[] = [] as  ScopeofReferalProduct[];
  }

File: CSharp.Functions/Common/Entities/FirmDetails/ProvidersAndIntroducers/IntroducerReferalProduct.cs
namespace Common.Entities
{
    public class IntroducerReferalProduct
    {
        public bool IsReferralsprovided { get; set; }
        public string? FixedFeePerReferral { get; set; }
        public string? FixedFeeAndAPercentageOfRevenue { get; set; }
        public string? RemunerationVaryByProduct { get; set; }
        public string? RemuneratedInitiallyOnRenewal { get; set; }
        public int MonthsNotice { get; set; }
        public List<ScopeofReferalProduct>? Products { get; set; }
    }
}
